### ISSUE TITLE
git lfs mac download issue

### ISSUE THREAD

**Issue by mfrank-gogo** [git-lfs/git-lfs#1561]
On macOS Sierra, using git-lfs/1.4.1 (GitHub; darwin amd64; go 1.7), i get the following error when executing a `git lfs clone <sshrepo>` from GitLab Enterprise Edition 8.12.1-ee 8a575a5. The clone proceeds without any issue for about 5min, then i get the following error:

```
Smudge error: Error downloading 9ebabebf729ff97f4cb5f2b39a0d50ab70e7e518f133f4a6056436ec1617f5c1: batch request: exit status 255: batch request: exit status 255
github.com/github/git-lfs/errors.newWrappedError
        /private/tmp/git-lfs-20160826-6787-1o15v20/git-lfs-1.4.1/src/github.com/github/git-lfs/errors/types.go:166
github.com/github/git-lfs/errors.NewSmudgeError
        /private/tmp/git-lfs-20160826-6787-1o15v20/git-lfs-1.4.1/src/github.com/github/git-lfs/errors/types.go:252
github.com/github/git-lfs/lfs.PointerSmudge
        /private/tmp/git-lfs-20160826-6787-1o15v20/git-lfs-1.4.1/src/github.com/github/git-lfs/lfs/pointer_smudge.go:69
github.com/github/git-lfs/lfs.(*Pointer).Smudge
        /private/tmp/git-lfs-20160826-6787-1o15v20/git-lfs-1.4.1/src/github.com/github/git-lfs/lfs/pointer.go:64
github.com/github/git-lfs/commands.smudgeCommand
        /private/tmp/git-lfs-20160826-6787-1o15v20/git-lfs-1.4.1/src/github.com/github/git-lfs/commands/command_smudge.go:66
github.com/github/git-lfs/vendor/github.com/spf13/cobra.(*Command).execute
        /private/tmp/git-lfs-20160826-6787-1o15v20/git-lfs-1.4.1/src/github.com/github/git-lfs/vendor/github.com/spf13/cobra/command.go:477
github.com/github/git-lfs/vendor/github.com/spf13/cobra.(*Command).Execute
        /private/tmp/git-lfs-20160826-6787-1o15v20/git-lfs-1.4.1/src/github.com/github/git-lfs/vendor/github.com/spf13/cobra/command.go:551
github.com/github/git-lfs/commands.Run
        /private/tmp/git-lfs-20160826-6787-1o15v20/git-lfs-1.4.1/src/github.com/github/git-lfs/commands/commands.go:70
main.main
        /private/tmp/git-lfs-20160826-6787-1o15v20/git-lfs-1.4.1/git-lfs.go:33
runtime.main
        /usr/local/Cellar/go/1.7/libexec/src/runtime/proc.go:183
runtime.goexit
        /usr/local/Cellar/go/1.7/libexec/src/runtime/asm_amd64.s:2086
```

this issue is reproducible, it seems to be a random file which causes the issue each time. Also reporoducible on various networks (wired, wireless, vpn). Cloning the same repository, with the same git-lfs version, on a linux machine is successful.

**Comment by ttaylorr** [comment]
Hi @mfrank-gogo, thanks for opening this. I have a hunch this a problem in the connection between `git-lfs-authenticate` and your GitLab instance. Could you run the same command as above with `GIT_TRACE=1` and paste the results? That will give us the arguments to run the `git-lfs-authenticate` command with, and we can debug from there. My hunch is that this is a credential expiration issue.

**Comment by mfrank-gogo** [comment]
Hi Taylor, I suspect you are right. I tried this with GIT_TRACE enabled, and I get tons of `trace git-lfs: HTTP: HTTP Basic: Access denied` errors for the downloads with both 401 and 403 response codes coming from gitlab. The retry looping in the lfs client does result in a successful authentication for some files, but, after the 182nd successful file download, I see:

`Git LFS: (182 of 1325 files) 743.46 MB / 3.83 GB                                                                                                                                                    trace git-lfs: xfer: adapter "basic" worker 0 finished job for "3aee6f8ba50d8a58e3c4e4ae923ec12c097203da1caefedfadc1270679bd10b0"
trace git-lfs: xfer: adapter "basic" End()
trace git-lfs: xfer: adapter "basic" worker 1 stopping
trace git-lfs: xfer: adapter "basic" worker 0 stopping
trace git-lfs: xfer: adapter "basic" worker 2 stopping
trace git-lfs: xfer: adapter "basic" stopped
Git LFS: (183 of 1325 files) 743.46 MB / 3.83 GB
Authentication required: Authorization error: https://git.gogoair.com/acpur/ACPU2.git/gitlab-lfs/objects/0c1043f00fd6fd67977064537731796a691ab5e770e09b04d7768a59f9ee9640
Check that you have proper access to the repository
Authentication required: Authorization error: https://git.gogoair.com/acpur/ACPU2.git/gitlab-lfs/objects/3a1be12511a91dc5aaf163c11b81c85d907920e7333b01c6d736041f98dc6c3b
Check that you have proper access to the repository`

The bottom line continues for the rest of the files and they don't get retried.

the `git-lfs-authenticate` arguments look like:
`trace git-lfs: ssh: git@git.gogoair.com git-lfs-authenticate acpur/ACPU2.git download`

**Comment by e4r7hbug** [comment]
Is there a way to make the error more clear without needing `GIT_TRACE=1`? I feel that a small exit error message would be easier to understand instead of an exit 255 with a stack trace.

**Comment by ttaylorr** [comment]
In this case, I'd rather have the output from `GIT_TRACE`, since it contains output that is not necessary for normal LFS usage, but contains key information for tracking down this bug.

@mfrank-gogo could you try again with the following:

```bash
$ GIT_TRACE=1 git lfs clone <repo> | grep "git-lfs-authenticate"
```

**Comment by mfrank-gogo** [comment]
Hi @ttaylorr , the output is below

```
$ export GIT_TRACE=1
$ git lfs clone git@git.gogoair.com:acpur/ACPU2.git 2>&1 | grep "git-lfs-authenticate"
trace git-lfs: ssh: git@git.gogoair.com git-lfs-authenticate acpur/ACPU2.git download
trace git-lfs: ssh: git@git.gogoair.com git-lfs-authenticate acpur/ACPU2.git download
trace git-lfs: ssh: git@git.gogoair.com git-lfs-authenticate acpur/ACPU2.git download
trace git-lfs: ssh: git@git.gogoair.com git-lfs-authenticate acpur/ACPU2.git download
trace git-lfs: ssh: git@git.gogoair.com git-lfs-authenticate acpur/ACPU2.git download
trace git-lfs: ssh: git@git.gogoair.com git-lfs-authenticate acpur/ACPU2.git download
trace git-lfs: ssh: git@git.gogoair.com git-lfs-authenticate acpur/ACPU2.git download
trace git-lfs: ssh: git@git.gogoair.com git-lfs-authenticate acpur/ACPU2.git download
trace git-lfs: ssh: git@git.gogoair.com git-lfs-authenticate acpur/ACPU2.git download
trace git-lfs: ssh: git@git.gogoair.com git-lfs-authenticate acpur/ACPU2.git download
trace git-lfs: ssh: git@git.gogoair.com git-lfs-authenticate acpur/ACPU2.git download
trace git-lfs: ssh: git@git.gogoair.com git-lfs-authenticate acpur/ACPU2.git download
trace git-lfs: ssh: git@git.gogoair.com git-lfs-authenticate acpur/ACPU2.git download
trace git-lfs: ssh: git@git.gogoair.com git-lfs-authenticate acpur/ACPU2.git download
trace git-lfs: ssh: git@git.gogoair.com git-lfs-authenticate acpur/ACPU2.git download
trace git-lfs: ssh: git@git.gogoair.com git-lfs-authenticate acpur/ACPU2.git download
trace git-lfs: ssh: git@git.gogoair.com git-lfs-authenticate acpur/ACPU2.git download
trace git-lfs: ssh: git@git.gogoair.com git-lfs-authenticate acpur/ACPU2.git download
trace git-lfs: ssh: git@git.gogoair.com git-lfs-authenticate acpur/ACPU2.git download
trace git-lfs: ssh: git@git.gogoair.com git-lfs-authenticate acpur/ACPU2.git download
trace git-lfs: ssh: git@git.gogoair.com git-lfs-authenticate acpur/ACPU2.git download
trace git-lfs: ssh: git@git.gogoair.com git-lfs-authenticate acpur/ACPU2.git download
trace git-lfs: ssh: git@git.gogoair.com git-lfs-authenticate acpur/ACPU2.git download
trace git-lfs: ssh: git@git.gogoair.com git-lfs-authenticate acpur/ACPU2.git download
trace git-lfs: ssh: git@git.gogoair.com git-lfs-authenticate acpur/ACPU2.git download
trace git-lfs: ssh: git@git.gogoair.com git-lfs-authenticate acpur/ACPU2.git download
trace git-lfs: ssh: git@git.gogoair.com git-lfs-authenticate acpur/ACPU2.git download
```

**Comment by ttaylorr** [comment]
Hi @mfrank-gogo is this still an issue? I am going to assume that it has since been resolved since the lack of noise on this thread. If this is incorrect, please do not hesitate to let me know here and I would be more than happy to take a look. Thanks!
